src/geo/map/cartodb-layer.js

```diff
@@ -136,6 +136,17 @@ var CartoDBLayer = LayerModelBase.extend({
     this.set('source', source, options);
   },
 
+  /**
+   * Check if an analysis node is the layer's source.
+   * Only torque and cartodb layers have a source otherwise return false.
+   */
+  hasSource: function (analysisModel) {
+    if (!(analysisModel instanceof AnalysisModel)) {
+      throw new TypeError('CartodbLayer.hasSource must be called with an "AnalysisModel" but got: ' + analysisModel);
+    }
+    return this.getSourceId() === analysisModel.get('id');
```

Comment:
Why not `return this.getSource() === analysisModel`? Another option could be to implement an `equals`  method on `AnalysisModel`:

```
return this.getSource().equals(analysisModel);
```

Reply:
> Why not return this.getSource() === analysisModel?

Strict equal compares object references and this can lead to errors if for some reason the object is cloned and since `ids` must be unique I think that comparing them will do the job.

Should I change the code and use `this.getSource === analysisModel` ?

Reply:
`return this.getSource().equals(analysisModel);` would be 👌 because we could probably reuse it in other places, but I leave it up to you at this point. PR is approved so these are only suggestions...